src/crafting.cpp

```diff
@@ -1113,8 +1113,15 @@ bool player::disassemble( item &obj, int pos, bool ground, bool interactive )
             list << "- " << elem.front().to_string() << std::endl;
         }
 
-        if( !query_yn( _( "Disassembling the %s may yield:\n%s\nReally disassemble?" ), obj.tname().c_str(),
-                       list.str().c_str() ) ) {
+        if ( !r.learn_by_disassembly.empty() && !knows_recipe( &r ) && can_decomp_learn( r ) )
```

Comment:
Should be like `if( foo ) {` : `if` is glued to the opening bracket; curly bracket is on the same line as `if`.